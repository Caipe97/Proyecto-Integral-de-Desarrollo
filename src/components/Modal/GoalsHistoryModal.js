import React, { useState } from 'react';
import Modal  from 'react-bootstrap/Modal';
import { Button } from 'react-bootstrap';
import GoalsSearchBar from '../SearchBar/GoalsSearchBar';

const GoalsHistoryModal =(props) =>{
  const [modalShow, setModalShow] = useState(false);
  return (
    <>
      <Button  type="button" variant="primary" style={{height: 50, width: "100%",backgroundColor: 'rgb(18, 207, 90)', borderColor: 'rgb(18, 207, 90)'}} onClick={() => setModalShow(true)}>
        Ver todas las metas
      </Button>
      
      <Modal
        show={modalShow}
        size="lg"
        aria-labelledby="contained-modal-title-vcenter"
        centered
      >
        <Modal.Header>
          <Modal.Title id="contained-modal-title-vcenter">
            {'Todas las Metas'}
          </Modal.Title>
          <Button type="button" onClick={() => setModalShow(false)} style={{ backgroundColor: 'white', borderColor: 'white', color: "black"}}>X</Button>
        </Modal.Header>
        <Modal.Body>
          <GoalsSearchBar {...props}/>
        </Modal.Body>
      </Modal>
    </>
  );
}
  
  export default GoalsHistoryModal
;