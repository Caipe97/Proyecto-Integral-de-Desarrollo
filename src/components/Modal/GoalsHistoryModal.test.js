import React from 'react';
import { shallow } from 'enzyme'; 
import GoalsHistoryModal from './GoalsHistoryModal';
//    <GoalsHistoryModal goal={row} history={this.props.history} onDeleteGoal={this.props.onDeleteGoal} onUpdateCurrentGoalInState={this.props.onUpdateCurrentGoalInState} />
          //props.goal.objectives.map
describe('GoalsHistoryModal tests', () => {
    const historyMock = { push: jest.fn() };
    const preventDefault = { preventDefault: jest.fn() };
    const mockPropsSignIn = {
        //goal: {name:'enero',dateStart:"2021-09-30T14:09:34.756Z",totalCalories:300,objectives:[]}, 
        preventDefault:jest.fn(),
        onGetMealsFromUser:jest.fn(),
        onDeleteMeal:jest.fn(),
        onUpdateCurrentMealInState:jest.fn(),
        onLogout:jest.fn(),
        onGetMealsByPeriod: jest.fn(),
        onGetFoodCategories: jest.fn(),
        onGetLastYearsMeals: jest.fn(),
        onGetGoalsFromUser: jest.fn(),
        onDeleteGoal: jest.fn(),
        onUpdateCurrentGoalInState:jest.fn(),
    };
    let wrapper;
    

    beforeEach(() => {
        wrapper = shallow(<GoalsHistoryModal {...mockPropsSignIn} history={historyMock}/>);
     
    })

    it('expect to render GoalsHistoryModal component to render', () => {
        expect(wrapper).toMatchSnapshot();
    })
    it('expect to change  onclick buttons for show more info',  () => {
         wrapper.find('[type="button"]').at(0).simulate('click');
    })
    it('expect to change  onclick buttons for show more info for hide this',  () => {
        wrapper.find('[type="button"]').at(1).simulate('click');
   })
   
})
