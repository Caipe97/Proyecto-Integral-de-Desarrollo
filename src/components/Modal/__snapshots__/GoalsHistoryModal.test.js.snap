// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoalsHistoryModal tests expect to render GoalsHistoryModal component to render 1`] = `
<Fragment>
  <Button
    active={false}
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "rgb(18, 207, 90)",
        "borderColor": "rgb(18, 207, 90)",
        "height": 50,
        "width": "100%",
      }
    }
    type="button"
    variant="primary"
  >
    Ver todas las metas
  </Button>
  <Modal
    animation={true}
    aria-labelledby="contained-modal-title-vcenter"
    autoFocus={true}
    backdrop={true}
    centered={true}
    dialogAs={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      }
    }
    enforceFocus={true}
    keyboard={true}
    restoreFocus={true}
    show={false}
    size="lg"
  >
    <ModalHeader
      closeButton={false}
      closeLabel="Close"
    >
      <ModalTitle
        id="contained-modal-title-vcenter"
      >
        Todas las Metas
      </ModalTitle>
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "white",
            "color": "black",
          }
        }
        type="button"
        variant="primary"
      >
        X
      </Button>
    </ModalHeader>
    <ModalBody>
      <GoalsSearchBar
        history={
          Object {
            "push": [MockFunction],
          }
        }
        onDeleteGoal={[MockFunction]}
        onDeleteMeal={[MockFunction]}
        onGetFoodCategories={[MockFunction]}
        onGetGoalsFromUser={[MockFunction]}
        onGetLastYearsMeals={[MockFunction]}
        onGetMealsByPeriod={[MockFunction]}
        onGetMealsFromUser={[MockFunction]}
        onLogout={[MockFunction]}
        onUpdateCurrentGoalInState={[MockFunction]}
        onUpdateCurrentMealInState={[MockFunction]}
        preventDefault={[MockFunction]}
      />
    </ModalBody>
  </Modal>
</Fragment>
`;
